src/services/job-state-checker.js

```diff
@@ -0,0 +1,49 @@
+const P = require('bluebird');
+const agent = require('superagent-promise')(require('superagent'), P);
+const jobDeserializer = require('../deserializers/job');
+const ProgressBar = require('progress');
+const { promisify } = require('util');
+const config = require('../config');
+
+const setTimeoutAsync = promisify(setTimeout);
+
+function JobStateChecker(message, logError) {
+  const bar = new ProgressBar(`${message} [:bar] :percent `, { total: 100 });
+  bar.update(0);
+
+  this.check = async (jobId, projectId) => {
+    try {
+      const jobResponse = await agent
+        .get(`${config.serverHost}/api/jobs/${jobId}`)
+        .set('Authorization', `Bearer ${config.authToken}`)
+        .set('forest-origin', 'Lumber')
+        .set('forest-project-id', projectId)
+        .then(response => jobDeserializer.deserialize(response.body));
+
+      if (jobResponse && jobResponse.state) {
+        let isBarComplete = false;
+        if (jobResponse.progress) {
+          bar.update(jobResponse.progress / 100);
+          isBarComplete = bar.complete;
+        }
+        if (jobResponse.state !== 'inactive' && jobResponse.state !== 'active') {
+          if (jobResponse.state === 'complete' && !isBarComplete) {
+            bar.update(1);
+          }
+          return jobResponse.state !== 'failed';
+        }
+      }
+    } catch (error) {
+      if (!error.status) {
+        throw error;
+      }
+
+      logError(`HTTP ${error.status}: ${error.message}`, { exit: 100 });
+    }
+
+    await setTimeoutAsync(1000);
+    return this.check(jobId, projectId);
+  };
+}
+
+module.exports = JobStateChecker;
```

Comment:
Moving this nice "refactoring" component in a dedicated PR would reduce the diff of this PR and make it lighter.

Reply:
So, do you want me to move it?